### ISSUE TITLE
Inconsistent Boolean comparison

### ISSUE THREAD

**Issue by zentol** [stratosphere/stratosphere#761]
The BooleanComparator.compare() works the opposite way of Boolean/BooleanValue.compareTo().

Consider the following piece of code:

```
public static void main(String[] args) throws IOException{
        Boolean x = true;
        Boolean y = false;
        System.out.println(x.compareTo(y));

        BooleanValue t = new BooleanValue(true);
        BooleanValue s = new BooleanValue(false);
        System.out.println(t.compareTo(s));

        BooleanSerializer bs = new BooleanSerializer();
        TestOutputView out1 = new TestOutputView();
        bs.serialize(true, out1);
        TestInputView in1 = out1.getInputView();
        TestOutputView out2 = new TestOutputView();
        bs.serialize(false, out2);
        TestInputView in2 = out2.getInputView();

        System.out.println(new BooleanComparator(true).compare(in1, in2));
    }
```

This prints

```
1
1
-1
```

intended?

**Comment by zentol** [comment]
the basicTypeComparator.compareToReference() is also odd. 

```
@Override
    public int compareToReference(TypeComparator<T> referencedComparator) {
        int comp = ((BasicTypeComparator<T>) referencedComparator).reference.compareTo(reference);
        return ascendingComparison ? comp : -comp;
    }
```

shouldnt it compare its own reference with the passed one instead of the other way around?

**Comment by markus-h** [comment]
I think everything should be consistent to the Boolean implementation of Java, so BooleanComparator and the basicTypeComparator seems to be wrong for me.

**Comment by StephanEwen** [comment]
The TypeComparators kind of swap the order of arguments internally. I think
all of the TypeComparators do it like that (but please verify that
statement).

If you want to change it, you would need to adjust every place where it is
invoked.

**Comment by zentol** [comment]
only the boolean comparator works like this.
exempt from the int comparator (i did check them all):

```
int i1 = firstSource.readInt();
int i2 = secondSource.readInt();
int comp = (i1 < i2 ? -1 : (i1 == i2 ? 0 : 1)); 
```

in contrast the boolean one:

```
final int fs = firstSource.readBoolean() ? 1 : 0;
final int ss = secondSource.readBoolean() ? 1 : 0;
int comp = ss - fs; 
```

**Comment by StephanEwen** [comment]
Okay, then it seems like a bug there. So, do the other comparators return a
consistent result in your sample test?

**Comment by zentol** [comment]
resolved in #764
